src/main/java/no/fintlabs/portal/oauth/NamOAuthClientService.java

```diff
@@ -89,11 +89,22 @@ public void removeOAuthClient(String clientId) {
 
     public OAuthClient getOAuthClient(String clientId) {
         log.info("Fetching client {}...", clientId);
-        try {
-            return restTemplate.getForObject(NamOAuthConstants.CLIENT_URL_TEMPLATE, OAuthClient.class, idpHostname, clientId);
-        } catch (Exception e) {
-            log.error("Unable to get client {}", clientId, e);
-            throw e;
+        for (int i = 1; true; i++) {
+            try {
+                return restTemplate.getForObject(NamOAuthConstants.CLIENT_URL_TEMPLATE, OAuthClient.class, idpHostname, clientId);
+            } catch (Exception e) {
+                log.error("Unable to get client {}, this was iteration number {}", clientId, i);
+
+                if (i == 10) {
+                    log.info("Maxed retries");
```

Comment:
Litt bedre melding. Forslag: "Failed to getOauthClient after max retry attempts. Giving up."